### DISCUSSION TITLE
Responsável técnico e CSRT -- 2025

### DISCUSSION THREAD

**Discussion by plinio-araujo** [TadaSoftware/PyNFe#371] [General]
O SEFAZ PR irá obrigar a tag (hashCSRT) nas emissões de NFe e NFCe. Existe alguma implementação para essa nova tag ?

O CSRT é um código de segurança gerado pela SEFAZ e fornecido à empresa responsável pelo sistema emissor de notas fiscais. Esse código, junto com a chave de acesso da NF-e, é utilizado para gerar o hashCSRT, uma assinatura digital que comprova a autenticidade do software emissor. Apenas a empresa desenvolvedora do sistema e a SEFAZ conhecem o CSRT, garantindo que o software utilizado para emitir as notas fiscais seja seguro e confiável.

Este hash será semelhante ao CSC para NFCe, onde será necessário incluir o hash criado para autorização da NF. 

Para a emissão do hash será necessário gerar um SHA-1 - base64 com a chave de acesso da NF e o CSRT disponibilizado pelo SEFAZ para o Software House. O resultado será o hashCSRT que será incluso junto com os dados do responsável técnico. 
Exemplo:
# Responsável técnico
    nota_fiscal.adicionar_responsavel_tecnico(
        cnpj='99999999000199',
        contato='TadaSoftware',
        email='tadasoftware@gmail.com',
        fone='11912341234',
        idCSRT = 01,
        hashCSRT ='aWv6LeEM4X6u4+qBI2OYZ8grigw='
      )

A previsão para obrigatoriedade deste hash para o responsável técnico é para abril/2025 no PR.

fonte:
https://sped.fazenda.pr.gov.br/NFe/Pagina/CSRT

**Comment by juniortada** [comment]
Criei uma issue e vou implementar essa feature:
https://github.com/TadaSoftware/PyNFe/issues/409

**Comment by plinio-araujo** [comment]
Perfeito, obrigado!
